test/functional/tag_controller_test.rb

```diff
@@ -148,7 +148,12 @@ def setup
     json = ActiveSupport::JSON.decode(@response.body)
 
     assert_not_nil json
-    # assert more things to check this
+    assert_equal tags(:spectrometer).nodes.first.body_preview,
+    json.first[:preview]
+    assert_equal tags(:spectrometer).nodes.first.main_image_url,
+    json.first[:image] 
```

Comment:
All right - there was a key error in the earlier test, and I managed to fix it -- unrelated to your work! Apologies! But it's great that we caught this. I'll push a new commit showing what happened and a fix.